internal/controller/backend_security_policy.go

```diff
@@ -290,3 +313,15 @@ func validateGCPCredentialsParams(gcpCreds *aigv1a1.BackendSecurityPolicyGCPCred
 
 	return nil
 }
+
+func getSecretNameAndOwnership(bsp *aigv1a1.BackendSecurityPolicy) (string, bool) {
+	if bsp.Spec.AzureCredentials != nil && bsp.Spec.AzureCredentials.ClientSecretRef != nil {
+		return string(bsp.Spec.AzureCredentials.ClientSecretRef.Name), false
+	}
+	if bsp.Spec.AWSCredentials != nil &&
+		bsp.Spec.AWSCredentials.CredentialsFile != nil &&
+		bsp.Spec.AWSCredentials.CredentialsFile.SecretRef != nil {
+		return string(bsp.Spec.AWSCredentials.CredentialsFile.SecretRef.Name), false
+	}
+	return rotators.GetBSPSecretName(bsp.Name), true
```

Comment:
Corrected your logic and used switch for readability & make it panic when we add new bsp type.

```suggestion
	switch bsp.Spec.Type {
	case aigv1a1.BackendSecurityPolicyTypeAWSCredentials:
		if bsp.Spec.AWSCredentials.OIDCExchangeToken != nil {
			return "", false
		}
	case aigv1a1.BackendSecurityPolicyTypeAzureCredentials:
		if bsp.Spec.AzureCredentials.OIDCExchangeToken != nil {
			return "", false
		}
	case aigv1a1.BackendSecurityPolicyTypeGCPCredentials:
	case aigv1a1.BackendSecurityPolicyTypeAPIKey:
		return "", false // APIKey does not require rotation.
	default:
		panic("BUG: unsupported backend security policy type." + string(bsp.Spec.Type))
	}
	return rotators.GetBSPSecretName(bsp.Name), true
```

Reply:
thanks for this. done